### PR TITLE
Add tag to mac prod and try builders to select bots without attached devices.

### PR DESCRIPTION
**Pull request by yusufm**
Add the "device_type:none" tag to mac prod and try builders that should run on bots that do not have a connected android or ios device.

Fixes: https://github.com/flutter/flutter/issues/75283

### CODE REVIEW

File: config/engine_config.star [hunk 1]
```diff
@@ -458,6 +458,7 @@ def engine_prod_config(platform_args, branch, version, ref, fuchsia_ctl_version)
         console_view_name = None,
         no_notify = True,
         priority = 30 if branch == "master" else 25,
+        dimensions = {"device_type": "none"},
```
Comment (godofredoc) [inline config/engine_config.star:461]: The drone dimensions are copied from the requesting task we don't need to add it here.

File: config/engine_config.star [hunk 2]
```diff
@@ -531,6 +532,7 @@ def engine_try_config(platform_args, fuchsia_ctl_version):
             gcs_goldens_bucket = "flutter_logs",
             no_lto = True,
         ),
+        dimensions = {"device_type": "none"},
```
Comment (godofredoc) [inline config/engine_config.star:535]: Can we add "mac_model": "Macmini8,1 here?

File: config/engine_config.star [hunk 3]
```diff
@@ -710,14 +712,15 @@ def engine_try_config(platform_args, fuchsia_ctl_version):
...
-        dimensions = {"mac_model": "Macmini8,1"},
+        dimensions = {"mac_model": "Macmini8,1", "device_type": "none"},
         **platform_args["mac"]
     )
     common.mac_try_builder(
         name = "Mac Engine Drone|drn",
         recipe = "engine_builder",
         repo = repos.ENGINE,
         list_view_name = list_view_name,
+        dimensions = {"device_type": "none"},
```
Comment (godofredoc) [inline config/engine_config.star:723]: Dimensions here are copied from main task, no need to pass it here.

**Review by godofredoc — COMMENTED** [review]
Do we also need to add the dimension to some of the devicelab_config.star tasks?

File: config/engine_config.star (hunk 1 above)
Comment (yusufm) [reply config/engine_config.star:461]: done

File: config/engine_config.star (hunk 2 above)
Comment (yusufm) [reply config/engine_config.star:535]: done

File: config/engine_config.star (hunk 3 above)
Comment (yusufm) [reply config/engine_config.star:723]: done

**Comment by yusufm** [comment]
> Do we also need to add the dimension to some of the devicelab_config.star tasks?

added them.

**Review by godofredoc — APPROVED** [review]

**Pull request merged**